src/binary.js

```diff
@@ -120,14 +120,46 @@ function getType(value) {
 }
 
 class Writer {
-    constructor(initialSize) {
-        this.bytes = new Uint8Array(initialSize || 100000000);
-        this.view = new DataView(this.bytes.buffer);
-        this.pos = 0;
+    constructor(chinkSize = 1 * 1024 * 1024 /* 1mb */) {
```

Comment:
1Mb for `chunkSize` is so much as for me, 64Kb is a good default. Moreover, it would be nice to add `options` to `encode()` to provide an ability to specify it